src/RelativeStream.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @see       http://github.com/zendframework/zend-diactoros for the canonical source repository
+ * @copyright Copyright (c) 2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-diactoros/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Diactoros;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Class RelativeStream
+ *
+ * Wrapper for default Stream class, representing subpart (starting from given offset) of initial stream.
+ * It can be used to avoid copying full stream, conserving memory.
+ * @example see Zend\Diactoros\AbstractSerializer::splitStream()
+ */
+class RelativeStream implements StreamInterface
+{
+    /**
+     * @var StreamInterface
+     */
+    private $decoratedStream;
+
+    /**
+     * @var int
+     */
+    private $offset;
+
+    /**
+     * Class constructor
+     *
+     * @param StreamInterface $decoratedStream
+     * @param int $offset
+     */
+    public function __construct(StreamInterface $decoratedStream, $offset)
```

Comment:
Would it make sense to also provide `$limit = null`? Or maybe am I being a feature creep here? /cc @weierophinney